src/ng-jest-transformer.ts

```diff
@@ -50,7 +51,7 @@ export class NgJestTransformer extends TsJestTransformer {
     filePath: Config.Path,
     transformOptions: TransformOptionsTsJest,
   ): TransformedSource | string {
-    const configSet = this._createConfigSet(transformOptions.config);
+    this.#configSet = this.#configSet ? this.#configSet : this._createConfigSet(transformOptions.config);
```

Comment:
Here if you use `this._configsFor` would allow `ts-jest` to do caching on our `NgJestConfig`. You can use `@ts-expect-error` with comment saying that `ts-jest` doesn’t expose the method as public typings.

Reply:
unfortunately it doesn't work properly

```
  TypeError: Cannot read property 'get' of undefined

       9 | export class NgJestCompiler extends TsCompiler {
      10 |   constructor(readonly configSet: ConfigSet, readonly jestCacheFS: Map<string, string>) {
    > 11 |     super(configSet, jestCacheFS);

```

anyway you know whats the problem and how to solve it, feel free to reject this PR and prepare correct solution, we already spend more time than you would spend doing that alone.. I need to get back to my primary work and looking forward for a fix. thanks in advance

Reply:
This is because the mocked configuration is missing the cache, opened https://github.com/thymikee/jest-preset-angular/pull/1318.

Reply:
the #1318 contains exactly what I was talking about :) thanks @sorin-davidoi